addons/website_event_meet/controllers/main.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from random import randint
+from werkzeug.exceptions import Forbidden, NotFound
+from werkzeug.utils import redirect
+
+from odoo import exceptions, http
+from odoo.http import request
+from odoo.addons.http_routing.models.ir_http import slug
+from odoo.osv import expression
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteEventMeetController(http.Controller):
+
+    def _get_event_rooms_base_domain(self, event):
+        search_domain_base = [
+            ('event_id', '=', event.id),
+        ]
+        if not request.env.user.has_group('event.group_event_user'):
+            search_domain_base = expression.AND([search_domain_base, [('is_published', '=', True)]])
+        return search_domain_base
+
+    # ------------------------------------------------------------
+    # MAIN PAGE
+    # ------------------------------------------------------------
+
+    @http.route(["/event/<model('event.event'):event>/meeting_rooms"], type="http",
+                auth="public", website=True, sitemap=True)
+    def event_meeting_rooms(self, event, lang=None, open_room_id=None):
+        """Display the meeting rooms of the event on the frontend side.
+
+        :param event: event for which we display the meeting rooms
+        :param lang: lang id used to perform a search
+        :param open_room_id: automatically open the meeting room given
+        """
+        if not event.can_access_from_current_website():
+            raise Forbidden()
+
+        return request.render(
+            "website_event_meet.event_meet",
+            self._event_meeting_rooms_get_values(event, lang=lang, open_room_id=open_room_id)
+        )
+
+    def _event_meeting_rooms_get_values(self, event, lang=None, open_room_id=None):
+        # meeting_rooms = event.sudo().meeting_room_ids.filtered(lambda m: m.active)
+        search_domain = self._get_event_rooms_base_domain(event)
+        meeting_rooms_all = request.env['event.meeting.room'].sudo().search(search_domain)
+        if lang:
+            search_domain = expression.AND([
+                search_domain,
+                [('room_lang_id', '=', int(lang))]
+            ])
+        meeting_rooms = request.env['event.meeting.room'].sudo().search(search_domain)
+        meeting_rooms = meeting_rooms.sorted(lambda m: (m.is_pinned, m.room_last_joined, m.id), reverse=True)
```

Comment:
as said before, maybe we could just use last_activity here instead of last_joined